pika/adapters/select_connection.py

```diff
@@ -27,20 +27,25 @@
 ERROR = 0x0008
 
 
-if pika.compat.PY2:
-    _SELECT_ERROR = select.error
-else:
-    # select.error was deprecated and replaced by OSError in python 3.3
-    _SELECT_ERROR = OSError
 
+_SELECT_ERRORS = (select.error, OSError, IOError)
```

Comment:
To guarantee consistency, `_SELECT_ERRORS` should be based on `_SELECT_ERROR_CHECKERS`. E.g.,

```
_SELECT_ERRORS = pika.compat.dictkeys(_SELECT_ERROR_CHECKERS)
```